client/src/Tasks/TaskRow.tsx

```diff
@@ -0,0 +1,147 @@
+                                {task.id}. {task.name}
+                            </Typography>
+                        </Grid>
+                        <Grid item>
+                            {task.errors > 0 && (
+                                <Chip
+                                    component="button"
+                                    onClick={() => {
+                                        if (expanded) {
+                                            setLogFilter(oldState =>
+                                                oldState === FILTER.ERRORS
+                                                    ? FILTER.ALL
+                                                    : FILTER.ERRORS,
+                                            );
+                                        }
+                                    }}
+                                    className={classes.error}
+                                    label={`${task.errors} ${
+                                        task.errors > 1 ? 'errors' : 'error'
+                                    }`}
+                                    size="small"
+                                />
+                            )}
+                            {task.warnings > 0 && (
+                                <Chip
+                                    component="button"
+                                    onClick={() => {
+                                        if (expanded) {
+                                            setLogFilter(oldState =>
+                                                oldState === FILTER.WARNINGS
+                                                    ? FILTER.ALL
+                                                    : FILTER.WARNINGS,
+                                            );
+                                        }
+                                    }}
+                                    className={classes.warning}
+                                    label={`${task.warnings} ${
+                                        task.warnings > 1
+                                            ? 'warnings'
+                                            : 'warning'
+                                    }`}
+                                    size="small"
+                                />
+                            )}
+                        </Grid>
+                    </Grid>
+                </PanelSummary>
+                <PanelDetails className={classes.panelDetails}>
+                    <Grid
+                        container
+                        direction="column"
+                        className={classes.panelDetails}
+                    >
+                        <Grid className={classes.logs}>
+                            {filteredLogs.map((l: Logs) => (
+                                <Typography
+                                    variant="body2"
+                                    color={dictionary[l.type]}
```

Comment:
No need for extra type in case of single use-case.
Also 'dictionary' name is little vague